### ISSUE TITLE
Feature request: improve logging

### ISSUE THREAD

**Issue by hsmade** [mibexsoftware/bamboo-plan-dsl-plugin#19]
Whenever I get an error when running the seed task, I just get a stack trace with almost no hint on what groovy script line actually caused it. Now I see 2 ways to improve this:
1) (have a way to enable) extensive logging on what part of the script is processed, so we can follow the flow and see where the stack trace was caused

or

2) Have a way to log the line number or snippet with the stack trace.

I don't know what possibilities you have in the code / in groovy handling, but it would surely help a lot. Even when I always use the DSL schemes for validation of what I throw at the plugin, I regularly get stack traces and need to strip down my config in order to find what caused the stack trace.

**Comment by mrueegg** [comment]
Hi,

Thanks for your feedback.

We currently provide error reporting with line numbers for basic syntax errors and not satisfied preconditions (like referencing a task which is not installed or a build variable that is not available). 

Additionally, we will try to provide debug logging for the next release so that it will be easier to see which part of the script was processed when an error occurs.

I'll report back here once we ship this.

Best regards,
Michael

**Comment by mrueegg** [comment]
We shipped a verbose logging option with release 1.5.3. Hope this helps :-)

**Comment by hsmade** [comment]
Just wondering, is the name of the artifact available when generating this error?

    error	20-Feb-2017 13:54:55	Plan DSL: task failed: Could not find referenced artifact. Have you shared this in your plan configuration?

**Comment by hsmade** [comment]
The actual issue that caused this error was a non-shared artifact, btw

**Comment by mrueegg** [comment]
The name is not yet available. But I will try to improve the error message for the next release.

If it's a non-shared artifact, have you tried to use an artifact in a job from a job in a different stage I guess?

**Comment by hsmade** [comment]
Correct, so the error is justified. But maybe you can find enough info at
that point in time to adjust it and hint at the sharing. Who knows :)
